### PR TITLE
[ML] Remove index.unassigned.node_left.delayed_timeout setting from M…

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
…L templates

This setting was introduced with the purpose of reducing the time took by
tests that shut nodes down. Tests like `MlDistributedFailureIT` and
`NetworkDisruptionIT`. However, it is unfortunate to have to set the value
to an explicit value in production. In addition, and most important, the dynamically
choosing the value for this setting makes it impossible to adopt static index template configs
that we register via `IndexTemplateRegistry`, which we need to use in order to start
registering ILM policies for the ML indices.

This commit removes this setting from our templates. I run the tests a few times and could
not see execution time differing significantly.

### CODE REVIEW

**Comment by droberts195** [comment]
I think we should add the setting for zero reassignment delay after a node leaves the cluster into the settings we use in internal cluster tests, i.e. adjacent to https://github.com/elastic/elasticsearch/blob/146f4ec15960c2f57abfbac8e68d7a200c145114/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/support/BaseMlIntegTestCase.java#L86

This will preserve the previous behaviour while still removing test logic from the production classes.

**Comment by dimitris-athanasiou** [comment]
@droberts195 Unfortunately it's not a node setting but an index setting. But CI bubbled up the tests that are affected. I'll set the setting in those tests.

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/BasicDistributedJobsIT.java
Comment (droberts195) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/BasicDistributedJobsIT.java]: Since both classes where this has been added extend `BaseMlIntegTestCase` it's probably worth putting this code in a utility method in that base class.

I also wonder if `NetworkDisruptionIT` might need it, depending on which random form of network disruption is chosen.  `NetworkDisruptionIT.testJobRelocation()` contains `ensureGreen()` calls.  It could be that for certain random number sequences these require a shard to reassign to attain green status.

**Review by droberts195 — APPROVED** [review]
I think you have unused imports, but LGTM when those are removed

**Pull request merged**
